### ISSUE TITLE
Macro values in rebar.config.script are assumed to be atoms

### ISSUE THREAD

**Issue by KrzysiekJ** [ninenines/erlang.mk#724]
According to [the documentation of `compile` module](http://erlang.org/doc/man/compile.html), macro name “is of type atom, and Value can be any term”. However the code introduced in d9a9158ccbb9 assumes that values are atoms and crashes in other cases. For example, [LFE in its `rebar.config.script` defines version as a list](https://github.com/rvirding/lfe/blob/71d8e5bafc59424bacd23c8af4b368d5a6db686c/rebar.config.script#L23-L43) and when it’s built as a dependency, it crashes with the following output:

```
{"init terminating in do_boot",{badarg,[{erlang,atom_to_list,["20.0"],[]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,670}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,469}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,470}]},{erl_eval,expr_list,6,[{file,"erl_eval.erl"},{line,878}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,422}]},{lists,foreach,2,[{file,"lists.erl"},{line,1338}]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,670}]}]}}
init terminating in do_boot ({badarg,[{erlang,atom_to_list,20.0,[]},{erl_eval,do_apply,6,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl_eval,expr_list,6,[{_},{_}]},{erl_eval,e

Crash dump is being written to: erl_crash.dump...done
```

**Comment by essen** [comment]
PR welcome.

**Comment by essen** [comment]
Seems this was fixed last year in c0f3338f. I see LFE is compiling without crashing for example, although it doesn't use this value at the moment. Please open a new ticket if there's still an issue.

Note that Erlang.mk will now load the lfe.mk plugin automatically when it sees the lfe-compile plugin for rebar3, and will also add lfe to the list of BUILD_DEPS (can be overriden in the parent project of course). Please experiment and tell me if that's a good idea and if it works.

Thanks!
